src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractExtensionMethodImportCompletionProvider.cs

```diff
@@ -0,0 +1,277 @@
+        public async Task<MultiDictionary<string, string>> GetPossibleExtensionMethodMatchesAsync(
+            Project currentProject,
+            ISet<string> targetTypeNames,
+            CancellationToken cancellationToken)
+        {
+            var solution = currentProject.Solution;
+            var graph = currentProject.Solution.GetProjectDependencyGraph();
+            var dependencies = graph.GetProjectsThatThisProjectTransitivelyDependsOn(currentProject.Id);
+            var relevantProjects = dependencies.Select(solution.GetProject)
+                                                 .Where(p => p.SupportsCompilation)
+                                                 .Concat(currentProject);
+
+            var results = new MultiDictionary<string, string>();
+
+            // Find matching extension methods from source.
+            foreach (var project in relevantProjects)
+            {
+                foreach (var document in project.Documents)
+                {
+                    var info = await document.GetSyntaxTreeIndexAsync(cancellationToken).ConfigureAwait(false);
+                    if (!info.ContainsExtensionMethod)
+                    {
+                        continue;
+                    }
+
+                    // Add simple extension methods with matching target type name
+                    foreach (var targetTypeName in targetTypeNames)
+                    {
+                        if (!info.SimpleExtensionMethodInfo.TryGetValue(targetTypeName, out var methodInfoIndices))
+                        {
+                            continue;
+                        }
+
+                        foreach (var index in methodInfoIndices)
+                        {
+                            if (info.TryGetDeclaredSymbolInfo(index, out var methodInfo))
+                            {
+                                results.Add(methodInfo.FullyQualifiedContainerName, methodInfo.Name);
+                            }
+                        }
+                    }
+
+                    // Add all complex extension methods, we will need to completely rely on symbols to match them.
+                    foreach (var index in info.ComplexExtensionMethodInfo)
+                    {
+                        if (info.TryGetDeclaredSymbolInfo(index, out var methodInfo))
+                        {
+                            results.Add(methodInfo.FullyQualifiedContainerName, methodInfo.Name);
+                        }
+                    }
+                }
+            }
+
+            // Find matching extension methods from metadata
+            foreach (var peReference in currentProject.MetadataReferences.OfType<PortableExecutableReference>())
+            {
+                var info = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+                    solution, peReference, loadOnly: false, cancellationToken).ConfigureAwait(false);
```

Comment:
same issues as before.  will this force creating these indices in the blocking completion path?

Reply:
Resolving this since the code has changed. Now we never force creating index on blocking path (expander path is not blocking, as far as I know)